### PR TITLE
WEEK-1 Build simple web-application

### PR DESCRIPTION
**Pull request by MaksTarnavskyi**
Create a simple web application and build a Docker image.

### CODE REVIEW

File: week1/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+USER appuser
+
+# set a directory for the app
+WORKDIR /usr/src/app
+
+# copy all the files to the container
+COPY flask-app .
+
+# install dependencies
+RUN pip install --no-cache-dir -r requirements.txt --user
```
Comment (truskovskiyk) [inline week1/Dockerfile:12]: let's install deps before 

```
# copy all the files to the container
COPY flask-app .
```

so we can really on cache

**Review by truskovskiyk — COMMENTED** [review]
@MaksTarnavskyi looks good, just one comment about caching

File: week1/Dockerfile (hunk 1 above)
Comment (MaksTarnavskyi) [reply week1/Dockerfile:12]: I thought we first need to copy the requirements file into docker and then install the deps.
I will check if it's possible to swap them

Comment (MaksTarnavskyi) [reply week1/Dockerfile:12]: image

Comment (MaksTarnavskyi) [reply week1/Dockerfile:12]: @truskovskiyk

Comment (MaksTarnavskyi) [reply week1/Dockerfile:12]: Or did I not understand something?

Comment (MaksTarnavskyi) [reply week1/Dockerfile]: I just removed --no-cache-dir flag, now everything is cached

File: week1/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+USER appuser
+
+# set a directory for the app
+WORKDIR /usr/src/app
+
+# copy all the files to the container
+COPY flask-app .
+
+# install dependencies
+RUN pip install -r requirements.txt --user
```
Comment (truskovskiyk) [inline week1/Dockerfile:12]: let's install des before coping all file to utilize cache: 

```
COPY flask-app/requirements.txt requirements.txt
RUN pip install -r requirements.txt --user
COPY flask-app .
```

https://stackoverflow.com/questions/34398632/docker-how-to-run-pip-requirements-txt-only-if-there-was-a-change

**Review by truskovskiyk — APPROVED** [review]
@MaksTarnavskyi looks good, please fix caching comment and good to merge!

**Pull request merged**